### PR TITLE
An 2143 nft models from streamline data

### PR DESCRIPTION
**Pull request by tarikceric**

### CODE REVIEW

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/silver__mint_actions2.sql [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    cluster_by = ['block_timestamp::DATE','_inserted_timestamp::DATE'],
+) }}
+
+WITH base_events AS (
+
+    SELECT
+        *
+    FROM
+        {{ ref('silver__events2') }}
+    WHERE block_timestamp :: date >= '2020-10-07'
```
Comment (desmond-hui) [inline models/silver/silver__mint_actions2.sql:14]: Where did this `block_timesamp` come from?

File: models/silver/nfts/silver__nft_mint_price_generic2.sql [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        ) 
+{% elif is_incremental() %}
+AND _inserted_timestamp >= (
+    SELECT
+        MAX(_inserted_timestamp)
+    FROM
+        {{ this }}
+)
+{% else %}
+AND block_timestamp :: date >= '2021-06-02'
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_mint_price_generic2.sql:42]: This needs to be a range

File: models/silver/nfts/silver__nft_mint_price_generic2.sql [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            LEAST(COALESCE(MAX(block_id), 81103515)+1,151386092)
+        FROM
+            {{ this }}
+        )
+        AND (
+        SELECT
+            LEAST(COALESCE(MAX(block_id), 81103515)+100000,151386092)
+        FROM
+            {{ this }}
+        )
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_mint_price_generic2.sql:69]: This should be block_timestamp

File: models/silver/nfts/silver__nft_sales_magic_eden_v1_2.sql [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  SELECT
+    MAX(_inserted_timestamp)
+  FROM
+    {{ this }}
+)
+{% else %}
+AND
+  e.block_timestamp :: date >= '2021-09-07'
+AND
+  t.block_timestamp :: date >= '2021-09-07'
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v1_2.sql:100]: These should be a range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+    SELECT
+        MAX(_inserted_timestamp)
+    FROM
+        {{ this }}
+)
+{% else %}
+    AND 
+        e.block_timestamp :: DATE >= '2022-01-08' -- no ME V2 contract before this date
+    AND 
+        t.block_timestamp :: DATE >= '2022-01-08'
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql:82]: these should be a range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+        )
+{% elif is_incremental() %}
+AND e._inserted_timestamp >= (
+    SELECT
+        MAX(_inserted_timestamp)
+    FROM
+        {{ this }}
+)
+{% else %}
+    AND e.block_timestamp :: DATE >= '2022-01-08' -- no ME V2 contract before this date
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql:172]: this should be a range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+{% elif is_incremental() %}
+AND
+    _inserted_timestamp >= (
+        SELECT
+            MAX(_inserted_timestamp)
+        FROM
+            {{ this }}
+    )
+{% else %}
+    AND e.block_timestamp:: DATE >= '2022-01-08' -- no ME V2 contract before this date
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql:229]: this should be a range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+{% elif is_incremental() %}
+WHERE
+    _inserted_timestamp >= (
+        SELECT
+            MAX(_inserted_timestamp)
+        FROM
+            {{ this }}
+    )
+{% else %}
+    WHERE block_timestamp :: DATE >= '2022-01-08' -- no ME V2 contract before this date
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql:285]: this should be a range

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+    AND e.block_timestamp :: DATE >= '2022-01-08' -- no ME V2 contract before this date
+{% endif %}
+),
+sellers AS (
+     SELECT
+        e.tx_id,
+        CASE WHEN signer <> instruction :accounts [1] :: STRING THEN 
+            instruction :accounts [6] :: STRING
+        ELSE 
+            instruction :accounts [1] :: STRING END AS seller
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql:181]: this logic was recently changed, make sure it matches the current table

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+            e.inner_instruction :instructions
+        ) > 1
+        LEFT OUTER JOIN TABLE(FLATTEN(inner_instruction :instructions)) i
+    WHERE
+        (
+            (
+                amount <> 0
+                AND inner_instruction_type = 'transfer'
+            )
+            OR inner_instruction_type = 'create'
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql:139]: This was also changed in the recent update.  See this commit https://github.com/FlipsideCrypto/solana-models/commit/cf75e9f70f6fb5581536076e753579756d2c890a

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql (hunk 8 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql]: added range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql (hunk 7 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql]: added range

File: models/silver/nfts/silver__nft_sales_magic_eden_v1_2.sql (hunk 4 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_magic_eden_v1_2.sql]: added range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql (hunk 6 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql]: added range

File: models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql (hunk 5 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_magic_eden_v2_2.sql]: added range

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**